src/StateService/StatePlugin.cs

```diff
@@ -15,27 +19,52 @@ public class StatePlugin : Plugin, IPersistencePlugin
         public override string Name => "StateService";
         public override string Description => "Enables MPT for the node";
 
-        private IActorRef store;
+        internal IActorRef Store;
+        internal IActorRef Validator;
 
         protected override void Configure()
         {
             Settings.Load(GetConfiguration());
+            RpcServerPlugin.RegisterMethods(this);
         }
 
         protected override void OnPluginsLoaded()
         {
-            store = System.ActorSystem.ActorOf(StateStore.Props(System, Settings.Default.Path));
+            Store = System.ActorSystem.ActorOf(StateStore.Props(System, this, Settings.Default.Path));
         }
 
         public override void Dispose()
         {
             base.Dispose();
-            System.EnsureStoped(store);
+            System.EnsureStoped(Store);
+            if (Validator != null) System.EnsureStoped(Validator);
         }
 
         void IPersistencePlugin.OnPersist(Block block, StoreView snapshot, IReadOnlyList<ApplicationExecuted> applicationExecutedList)
         {
             StateStore.Singleton.UpdateLocalStateRoot(block.Index, snapshot.Storages.GetChangeSet().Where(p => p.State != TrackState.None).ToList());
         }
+
+        [RpcMethod]
+        public JObject VoteStateRoot(JArray _params)
+        {
+            uint height = uint.Parse(_params[0].AsString());
+            int validator_index = int.Parse(_params[1].AsString());
+            byte[] sig = _params[2].AsString().HexToBytes();
+            Validator?.Tell(new Vote(height, validator_index, sig));
+            return true;
+        }
+
+        [ConsoleCommand("start validate", Category = "StateService", Description = "Start as a state validator if wallet is open")]
+        private void OnStartValidate()
+        {
+            var wallet_provider = GetService<IWalletProvider>();
+            if (wallet_provider is null)
```

Comment:
It can't be null, but when you call `GetWallet` it can be

Reply:
Done.